src/main/java/org/apache/mesos/hdfs/Scheduler.java

```diff
@@ -452,7 +454,7 @@ private boolean tryToLaunchNameNode(SchedulerDriver driver, Offer offer) {
     List<String> deadNameNodes = persistentState.getDeadNameNodes();
 
     if (deadNameNodes.isEmpty()) {
-      if (liveState.getNameNodeSize() == HDFSConstants.TOTAL_NAME_NODES) {
+      if (persistentState.getNameNodes().size() == HDFSConstants.TOTAL_NAME_NODES) {
```

Comment:
Hm, we actually moved this to be stored in PersistentState instead of LiveState. This makes sense because we do need to persist this for reconciliation. We already store the taskid and hostname in the persistent state. We are simply adding the taskname here.